src/main/docs/guide/service-discovery.adoc

```diff
@@ -27,6 +27,142 @@ In your HTTP client, you can use `my-service` as Service ID: `@Client("my-servic
+
+[%header, cols=2*]
+|===
+| Key
+| Description
+
+| `name`
+| name of the resource in Kubernetes in case it is different than the Service ID
+
+| `namespace`
+| namespace of the resource in case it's different than the configured namespace
+
+| `port`
+| port name in case the target resource is a https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services[Multi-Port Service]
+
+| `mode`
+| service specific <<Service discovery modes,discovery mode>> in case it's different than the globally configured discovery mode
+|===
+
+### Examples of service configurations
+#### Multi-port service
+For the following https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services[Multi-Port Service]:
+[source, yaml]
+.`my-service.yml`
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  selector:
+    app: MyApp
+  ports:
+    - name: http
+      protocol: TCP
+      port: 80
+      targetPort: 9376
+    - name: https
+      protocol: TCP
+      port: 443
+      targetPort: 9377
+----
+
+the manual service configuration for http port will be:
+[source, yaml]
+.`bootstrap.yml`
+----
+kubernetes:
+  client:
+    discovery:
+      services:
+      - my-service:
+          port: http
+----
+
+#### Headless service with selector
+For the following https://kubernetes.io/docs/concepts/services-networking/service/#headless-services[Headless service] with selector:
+[source, yaml]
+.`my-service.yml`
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  clusterIP: None
+  selector:
+    app: MyApp
+  ports:
+    - name: http
+      protocol: TCP
+      port: 80
+      targetPort: 9376
+----
+the manual service configuration will be:
+[source, yaml]
+.`bootstrap.yml`
+----
+kubernetes:
+  client:
+    discovery:
+      services:
+      - my-service:
+          mode: endpoint
+----
+
+#### ExternalName service type
+For the following https://kubernetes.io/docs/concepts/services-networking/service/#externalname[ExternalName service]:
+[source, yaml]
+.`my-service.yml`
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  namespace: prod
+spec:
+  type: ExternalName
+  externalName: launch.micronaut.io
+----
+the manual service configuration will be:
+[source, yaml]
+.`bootstrap.yml`
+----
+kubernetes:
+  client:
+    discovery:
+      services:
+      - my-service:
+          mode: service
+----
+
+## Service discovery modes
+Service discovery mode is a mechanism that allows to support different strategies for the actual service discovery in Kubernetes by implementing link:{sourcedir}/ kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/discovery/KubernetesServiceInstanceProvider.java[KubernetesServiceInstanceProvider] interface.
+
+Currently Micronaut Kubernetes implements two discovery modes:
+// todo - benefits
+* `endpoint` uses the Kubernetes `Endpoins` API
```

Comment:
not what it means for every mode:
- endpoint - client load balancing
- service - k8s load balancing